File: 프로그래머스/0/120897. 약수 구하기/약수 구하기.java
import java.util.*;
class Solution {
    public int[] solution(int n) {
        int[] answer = {};
        int count = 0;
        
        for(int i = 1; i <= n; i++) if(n % i == 0) count++;
        
        answer = new int[count];
        count = 0;
        
        for(int j = 1; j <= n; j++) {
            if(n % j == 0) {
                answer[count] = j;
                count++;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/0/120897. 약수 구하기/약수 구하기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(int n) {
    int* answer = (int*)malloc(n * sizeof(int));
    int idx = 0;
    
    for(int i = 1; i <= n; i++){
        if(n % i == 0){
            answer[idx++] = i;
        }
    }
    
    answer = realloc(answer, idx * sizeof(int));
    
    return answer;
}